src/ResourceManager/DeviceProvisioningServices/Commands.DeviceProvisioningServices/Cmdlet/IotDps/NewAzureRmIotDeviceProvisioningService.cs

```diff
@@ -0,0 +1,132 @@
+            Mandatory = false,
+            HelpMessage = "IoT Device Provisioning Service Allocation policy")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(new string[] { "Hashed", "GeoLatency", "Static" }, IgnoreCase = true)]
+        public string AllocationPolicy { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Sku")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(new string[] { "S1" }, IgnoreCase = true)]
+        public string SkuName { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ShouldProcess(Name, DPSResources.AddDeviceProvisioningService))
+            {
+                try
+                {
+                    if (string.IsNullOrEmpty(this.Location))
+                    {
+                        ResourceGroup resourceGroup = ResourceManagementClient.ResourceGroups.Get(this.ResourceGroupName);
+                        this.Location = resourceGroup.Location;
+                    }
+
+                    var provisioningServiceDescription = new ProvisioningServiceDescription()
+                    {
+                        Location = this.Location,
+                        Properties = new IotDpsPropertiesDescription(),
+                        Sku = new IotDpsSkuInfo()
+                    };
+
+                    if (this.AllocationPolicy != null)
+                    {
+                        PSAllocationPolicy psAllocationPolicy;
+                        if (Enum.TryParse<PSAllocationPolicy>(this.AllocationPolicy, true, out psAllocationPolicy))
+                        {
+                            provisioningServiceDescription.Properties.AllocationPolicy = psAllocationPolicy.ToString();
+                        }
+                        else
+                        {
+                            throw new ArgumentException("Invalid Allocation Policy");
+                        }
+                    }
+
+                    if (this.SkuName != null)
+                    {
+                        PSIotDpsSku psIotDpsSku;
+                        if (Enum.TryParse<PSIotDpsSku>(this.SkuName, true, out psIotDpsSku))
+                        {
+                            provisioningServiceDescription.Sku.Name = psIotDpsSku.ToString();
+                        }
+                        else
+                        {
+                            throw new ArgumentException("Invalid Sku");
+                        }
+                    }
+                    else
+                    {
+                        provisioningServiceDescription.Sku = new IotDpsSkuInfo(IotDpsSku.S1);
+                    }
+
+                    IotDpsCreateOrUpdate(this.ResourceGroupName, this.Name, provisioningServiceDescription);
+                    this.WriteObject(IotDpsUtils.ToPSProvisioningServiceDescription(GetIotDpsResource(this.ResourceGroupName, this.Name)), false);
+                }
+                catch (Exception e)
```

Comment:
To capture unwanted exception, so that it will return the error to the user.

Reply:
@anusapan this shouldn't be necessary since we do error-handling in the base [`AzurePSCmdlet`](https://github.com/Azure/azure-powershell/blob/preview/src/Common/Commands.Common/AzurePSCmdlet.cs#L634-L645) class